apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/CommonQueries.java

```diff
@@ -0,0 +1,202 @@
+package gov.cms.bfd.server.war.commons;
+
+import ca.uhn.fhir.rest.server.exceptions.ResourceNotFoundException;
+import ca.uhn.fhir.rest.server.exceptions.UnclassifiedServerFailureException;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import gov.cms.bfd.model.rif.entities.Beneficiary;
+import gov.cms.bfd.model.rif.entities.Beneficiary_;
+import gov.cms.bfd.server.sharedutils.BfdMDC;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.persistence.EntityManager;
+import javax.persistence.NoResultException;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.JoinType;
+import javax.persistence.criteria.Root;
+
+/** Common database queries that can be shared between V1 and V2. */
+public class CommonQueries {
+
+  /**
+   * Database function that checks all claims for a given beneficiaryId and returns a bitwise mask
+   * value that shows if a given claim type will have any data.
+   */
+  public static final String CHECK_CLAIMS_FOR_DATA_SQL =
+      "SELECT * FROM check_claims_mask(:beneIdValue)";
+
+  /**
+   * Database function that takes in a patient identifier type and value and returns a bigint
+   * BENE_ID if the identifier can find a beneficiaries record. Supported identifier types are: mbi,
+   * mbi-hash, hicn-hash.
+   */
+  public static final String FIND_BENE_ID_FROM_IDENTIFIER_SQL =
+      "SELECT * FROM find_beneficiary(:searchIdType, :searchIdValue)";
+
+  /** preclude construction from outsiders. */
+  private CommonQueries() {}
+
+  /**
+   * Query database to determine which claim types have data for the specified beneficiary. The
+   * database function returns a bitwise mask value that denotes which claim types will have data
+   * for the specified beneficiaryId. This represents a fast and efficient way to ignore a requested
+   * claim type that ultimately has no data for our beneficiaryId.
+   *
+   * <p>The database function returns bits as follows: CARRIER_CLAIMS : bit 0 INPATIENT CLAIMS : bit
+   * 1 OUTPATIENT CLAIMS : bit 2 SNF CLAIMS : bit 3 DME CLAIMS : bit 4 HHA CLAIMS : bit 5 HOSPICE
+   * CLAIMS : bit 6 PART D CLAIMS : bit 7
```

Comment:
I'm assuming the formatted mangled this. Could we avoid the formatter butchery with an unordered list?